src/preprocessor/tokenizer.py

```diff
@@ -0,0 +1,75 @@
+import tiktoken
+
+import torch
+from torch.utils.data import Dataset, DataLoader
+
+class Tokenizer:
+    def __init__(self, raw_text=None, batch_size=4, max_length=256,
+                    stride=128, output_dim=256):
+        self.batch_size = batch_size
+        self.max_length = max_length
+        self.stride     = stride
+        self.output_dim = output_dim
+
+        # Initialize the tokenizer
+        self.tokenizer = tiktoken.get_encoding("gpt2")
```

Comment:
is gpt2 our only option? - if other options are available parameterize the string "gpt2" into the constructor